src/ln/functional_tests.rs

```diff
@@ -5353,17 +5353,39 @@ fn test_update_fulfill_htlc_bolt2_after_malformed_htlc_message_must_forward_upda
 	check_added_monitors!(nodes[1], 1);
 }
 
+#[test]
+#[should_panic]
+fn test_commitment_signed_bolt2_sending_node_must_not_send_commitment_signed_no_updates() {
+	//BOLT 2 Requirement: A sending node:
+	//    * MAY send a commitment_signed message that only alters the fee.
+
+	let nodes = create_network(2);
+	let chan = create_announced_chan_between_nodes(&nodes, 0, 1);
+	let channel_id = chan.2;
+
+	let mut chan_state_lock = nodes[0].node.channel_state.lock().unwrap();
+	let chan_state = chan_state_lock.borrow_parts();
+
+	if let Some(mut chan) = chan_state.by_id.get_mut(&channel_id) {
+		let _result = chan.send_commitment();
```

Comment:
nit: Could be only chan.send_commitment()

Reply:
This emits a warning `unused `std::result::Result` that must be used` - Not sure which is better, I'll be changing the code to return a `ChannelError` anyway so I'll end up using the result :)

Reply:
Oh move the value out of the option to get off the warning, i.e unwrap()